integration-tests/tests/snapd_1_0_packages_test.go

```diff
@@ -94,17 +94,22 @@ func (s *snapd10PackagesTestSuite) postInteractions() apiInteractions {
 		payload:     s.snapPath,
 		waitPattern: `(?U){.*,"status":"active".*"status":"OK","status_code":200,"type":"sync"}`,
 		waitFunction: func() (string, error) {
-			output, err := genericRequest(s.resource()+"/"+data.BasicConfigSnapName+".sideload", "GET", nil)
+			output, err := genericRequest(&requestOptions{
+				resource: s.resource() + "/" + data.BasicConfigSnapName + ".sideload",
+				verb:     "GET"})
 			return string(output), err
 		}}}
 }
 
 func (s *snapd10PackagesTestSuite) putInteractions() apiInteractions {
 	return []apiInteraction{{
-		payload:     `{"` + data.BasicConfigSnapName + ".sideload" + `": "key: value"}`,
+		// this payload is adapted to the httpie client
+		payload:     data.BasicConfigSnapName + `.sideload:="{key: value}"`,
 		waitPattern: `(?Us){"result":.*` + data.BasicConfigSnapName + `.*key: value.*","status":"OK","status_code":200,"type":"sync"}`,
 		waitFunction: func() (string, error) {
-			output, err := genericRequest(s.resource()+"/"+data.BasicConfigSnapName+".sideload/config", "GET", nil)
+			output, err := genericRequest(&requestOptions{
+				resource: s.resource() + "/" + data.BasicConfigSnapName + ".sideload/config",
+				verb:     "GET"})
```

Comment:
Same, on the inner, and the outer struct/slice. Don't consider this a blocker for your branch, though.